### PR TITLE
RELEASE.md: update release instructions

### PR DESCRIPTION
**Pull request by fpetkovski**
## Description

Updates the instructions for cutting new releases.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -108,9 +112,3 @@ For patch releases, submit a pull request to merge back the release branch into
 ## Update website
 
 Bump the operator's version in the [website](https://github.com/prometheus-operator/website/blob/main/data/prometheusOperator.json) repository.
-
-## Update kube-prometheus
-
-Bump the versions of `github.com/prometheus-operator/prometheus-operator` in [prometheus-operator/kube-prometheus](https://github.com/prometheus-operator/kube-prometheus) by triggering the ["Upgrade to latest versions"](https://github.com/prometheus-operator/kube-prometheus/actions/workflows/versions.yaml) workflow.
-
-Take a breath. You're done releasing.
```
Comment (paulfantom) [inline RELEASE.md:116]: Let's bring this line back :)

**Review by paulfantom — COMMENTED** [review]
lgtm, just one nit

**Review by paulfantom — APPROVED** [review]
lgtm

**Comment by fpetkovski** [comment]
@paulfantom could you ptal again? I had to fix some CI jobs

**Pull request merged**